Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -177,6 +180,34 @@ void NetKDRequestDevice::Update()
   }
 }
 
+void NetKDRequestDevice::LogError(ErrorType error_type, s32 error_code)
+{
+  s32 new_code{};
+  switch (error_type)
+  {
+  case ErrorType::ACCOUNT:
+    new_code = -(101200 - error_code);
+    break;
+  case ErrorType::CLIENT:
+    new_code = -(107300 - error_code);
+    break;
+  case ErrorType::KD_DOWNLOAD:
+    new_code = -(107200 - error_code);
+    break;
+  case ErrorType::SERVER:
+    new_code = -(117000 + error_code);
+    break;
+  }
+
+  std::lock_guard lg(m_scheduler_buffer_lock);
+
+  m_scheduler_buffer[32 + (m_error_count & 0x1f)] = Common::swap32(new_code);
```

Comment:
```suggestion
  m_scheduler_buffer[32 + (m_error_count % 32] = Common::swap32(new_code);
```
IMO that's a clearer way to express the intended behaviour ("reset the index back to 0 once there are 32 errors but keep the error count").